speech/norm/__init__.py

```diff
@@ -231,42 +182,244 @@ def _email_handler(email: str) -> str:
+            as that should be handled by caller.
+        """
+        # Get first fullmatch from date regexes
+        m = next(
+            filter(
+                lambda x: x is not None,
+                (r.fullmatch(txt) for r in _DATE_REGEXES),
+            ),
+            None,  # Default if no matches are found
+        )
+        assert m is not None, f"Incorrect date format specified for date handler: {txt}"
+
+        # Handle 'DD/MM/YYYY' or 'MM. jan/feb/... [year]' match
+        gd = m.groupdict()
+        day = number_to_ordinal(gd["day"], gender="kk", case=case, number="et")
+        # Month names don't change in different declensions
+        month = (
+            _MONTH_NAMES[int(gd["month"]) - 1]  # DD/MM/YYYY specification
+            if gd["month"].isdecimal()
+            else _MONTH_NAMES[_MONTH_ABBREVS.index(gd["month"][:3])]  # Non-decimal
+        )
+        return (
+            f"{day} {month} {year_to_text(gd['year'])}"
+            if gd["year"]
+            else f"{day} {month}"
+        )
+
+    @classmethod
+    def timespan(cls, seconds: str) -> str:
+        """Voicify a span of time, specified in seconds."""
+        # TODO: Replace time_period_desc in queries/util/__init__.py
+        raise NotImplementedError()
+
+    @classmethod
+    def year(cls, txt: str, *, after_christ: Optional[str] = None):
+        """Voicify a year."""
+        ac = after_christ is not None and after_christ == "True"
+        return year_to_text(txt, after_christ=ac)
+
+    @classmethod
+    def years(cls, txt: str, *, after_christ: Optional[str] = None):
+        """Voicify text containing multiple years."""
+        ac = after_christ is not None and after_christ == "True"
+        return years_to_text(txt, after_christ=ac)
+
+    # Pronunciation of character names in Icelandic
+    _CHAR_PRONUNCIATION = {
+        "a": "a",
+        "á": "á",
+        "b": "bé",
+        "c": "sé",
+        "d": "dé",
+        "ð": "eð",
+        "e": "e",
+        "é": "é",
+        "f": "eff",
+        "g": "gé",
+        "h": "há",
+        "i": "i",
+        "í": "í",
+        "j": "joð",
+        "k": "ká",
+        "l": "ell",
+        "m": "emm",
+        "n": "enn",
+        "o": "o",
+        "ó": "ó",
+        "p": "pé",
+        "q": "kú",
+        "r": "err",
+        "s": "ess",
+        "t": "té",
+        "u": "u",
+        "ú": "ú",
+        "v": "vaff",
+        "w": "tvöfaltvaff",
+        "x": "ex",
+        "y": "ufsilon",
+        "ý": "ufsilon í",
+        "þ": "þoddn",
+        "æ": "æ",
+        "ö": "ö",
+        "z": "seta",
     }
-)
+
+    @classmethod
+    def spell(cls, txt: str):
+        """Spell out a sequence of characters."""
+        if not txt:
+            return ""
+        t = [cls._CHAR_PRONUNCIATION.get(c.lower(), c) if c != " " else "" for c in txt]
+        s = ", ".join(t).strip()
+        return re.sub(r"\s\s+", r" ", s)  # Shorten repeated whitespace
```

Comment:
The `re.sub()` is probably not needed here

Reply:
True, fixed.